src/Campaigns/Blocks/CampaignDonorsBlock/edit.tsx

```diff
@@ -0,0 +1,112 @@
+import {InspectorControls, useBlockProps} from '@wordpress/block-editor';
+import {BlockEditProps} from '@wordpress/blocks';
+import {
+    __experimentalNumberControl as NumberControl,
+    PanelBody,
+    SelectControl,
+    TextControl,
+    ToggleControl,
+} from '@wordpress/components';
+import {__} from '@wordpress/i18n';
+import ServerSideRender from '@wordpress/server-side-render';
+import {CampaignSelector} from '../shared/components/CampaignSelector';
+import useCampaign from '../shared/hooks/useCampaign';
+
+export default function Edit({
+    attributes,
+    setAttributes,
+}: BlockEditProps<{
+    campaignId: number;
+    showAnonymous: boolean;
+    showCompanyName: boolean;
+    showAvatar: boolean;
+    showButton: boolean;
+    donateButtonText: string;
+    sortBy: string;
+    donorsPerPage: number;
+    loadMoreButtonText: string;
+}>) {
+    const blockProps = useBlockProps();
+    const {campaign, hasResolved} = useCampaign(attributes.campaignId);
+
+    const {
+        showAnonymous,
+        showCompanyName,
+        showAvatar,
+        showButton,
+        donateButtonText,
+        sortBy,
+        donorsPerPage,
+        loadMoreButtonText,
+    } = attributes;
+
+    return (
+        <div {...blockProps}>
+            <CampaignSelector attributes={attributes} setAttributes={setAttributes}>
+                <ServerSideRender block="givewp/campaign-donors-block" attributes={attributes} />
+            </CampaignSelector>
+
+            {hasResolved && campaign && (
```

Comment:
I'm not sure that the returned result would be null for the campaign. It might be better to do something like campaign?.id.

Reply:
Modded on da79d91796308727f67d0a5cb0e347092378f467